### ISSUE TITLE
Autoformatting: nested if-else indentation

### ISSUE THREAD

**Issue by vers-one** [dotnet/roslyn#12315]
**Version Used**: Visual Studio 2015 Update 3

**Steps to Reproduce**:
1. Paste the following code:

```
void Test()
{
    if (1 == 1)
        Test();
    else
        if (2 == 2)
            Test();
        else
            Test();
}
```

**Expected Behavior**: autoformatting like in previous versions of Visual Studio (proper indentation of nested if-else blocks).

**Actual Behavior**: Visual Studio 2015 currently formats it like this:

```
void Test()
{
    if (1 == 1)
        Test();
    else
        if (2 == 2)
        Test();
    else
        Test();
}
```

**Comment by Pilchie** [comment]
I've looked at this issue before (It was also reported at http://connect.microsoft.com/VisualStudio/feedback/details/1711675/autoformatting-issues).  Sadly it turns out it's actually quite difficult to fix in our current formatting engine :(

We haven't spent a lot of time trying, since it's *far* more common to format the "else" and "if" on the same line:

```C#
void Test()
{
    if (1 == 1)
        Test();
    else if (2 == 2)
        Test();
    else
        Test();
}
```

or to use braces:

```C#
void Test()
{
    if (1 == 1)
        Test();
    else
    {
        if (2 == 2)
            Test();
        else
            Test();
    }
}
```

**Comment by flipchart** [comment]
@Pilchie Sorry to hijack the issue, but formatting for me is a major pain. Is this the right place to be reporting formatting issues (not just regressions from updates, but issues for several versions)? Will they be looked at with the current state of the formatter (as per your comment?

**Comment by Pilchie** [comment]
This *is* the right place for reporting C# and VB formatting issues, and yes, we will look at them.

However, I can't promise we will fix every issue as there a LOT of possible ways to format code, and some are barely used.

**Comment by DustinCampbell** [comment]
Just received a Visual Studio Feedback issue for this, so +1

**Comment by mindave** [comment]
The biggest issue with this 'bug' is that when editing an existing file that has never been edited in VS2015 and use ^K+^D, it messes up good formatting throughout the file.  :( 

> We haven't spent a lot of time trying, since it's far more common to format 
> the "else" and "if" on the same line

Sorry, this is just a 'cop out'!?  No one in our organization **ever** uses that format and we've been coding in C since the early '80s

> or to use braces:
Absolutely loathe to insert 'unnecessary' braces, usually I'll take the time to remove them when I find them in code.

Code formatting is all about readability and both of these 'solutions' do not make for more readable code...

**Comment by versfx** [comment]
This can be reproduced in Visual Studio 2017 RC as well.

**Comment by Pilchie** [comment]
Also reported at https://developercommunity.visualstudio.com/content/problem/42991/c-if-nesting.html

**Comment by SOHODeveloper** [comment]
" Sadly it turns out it's actually quite difficult to fix in our current formatting engine :("
Well, that just tells me that your engine is of bad design.  There are NUMEROUS bad/incorrect formatting issues in VS 2015 and 2017 that all worked correctly prior to 2015, so we KNOW it can be done correctly.  
As mindave said, "Code formatting is all about readability and both of these 'solutions' do not make for more readable code..."

I spend more of my day formatting code than writing code.  That is NOT productivity.

**Comment by CyrusNajmabadi** [comment]
Fixed with https://github.com/dotnet/roslyn/pull/81178
